### ISSUE TITLE
Block elements converted to inline should retain isolation

### ISSUE THREAD

**Issue by r12a** [w3c/alreq#193]
@ntounsi describes some interesting behaviour that i will reproduce here:

>  `div` (or others block elements) default to isolate only in FF. Implementation problem.
> Following the example in this thread,
> 
> `<div>ابجد</div>` 
> `<div style="background-color: yellow;" >2 abc</div>`
> with the style `div {display: inline;}`
> 
> gives (yellow for readability):
> ![isolate1](https://user-images.githubusercontent.com/15004708/44160883-a99c8f00-a0b3-11e8-8272-bd7ec8e4c6b5.png) in FF (isolation)
> ![isolate2](https://user-images.githubusercontent.com/15004708/44160895-b4efba80-a0b3-11e8-8811-1826ead5e3e3.png)  in Safari, Chrome and Opera
> 
> The 2 sticks to Arabic (RTL) in the last image, due to non isolation.

**Comment by r12a** [comment]
I would have expected that the bidi isolation that applies to an HTML `div` or `li` when it is a block element would remain unchanged around the content of the element when CSS is used to convert it to inline, eg. for constructing tabs on a page.  (I would expect that the same applies for all block elements that are defined in HTML to have isolation.)

I coudn't find any text in the Writing Modes spec to confirm that expectation.  @fantasai @kojiishi what do you think?

**Comment by Zhang-Junzhi** [comment]
Here is the recommended default UA stylesheet in HTML spec:

https://html.spec.whatwg.org/#bidi-rendering

Firefox seems to be the reasonable one, and also it follows the recommendation of the spec.

**Comment by kojiishi** [comment]
Right, the spec is in HTML. We have a contributor on [our crbug 296863](https://crbug.com/296863) but he is not active these days.

**Comment by Zhang-Junzhi** [comment]
And that was the reason why I argued that the HTML spec's specifying div as meaningless at all is inaccurate. Because the `unicode-bidi` value for a div defaults to `isolate`, which is inconsistent with specifying div as meaningless at all

**Comment by matial** [comment]
It seems to me that isolation for a block is irrelevant until it is converted to inline. Thus FF's behavior (isolating the converted block) is the more useful one.
Also, when in doubt, isolating is less likely to produce surprising effects than not isolating, so if the specs need to be clarified, this is the preferred choice.
